test/Test/Query.hs

```diff
@@ -0,0 +1,119 @@
+module Test.Query
+    ( querySpecs
+    ) where
+
+import Data.Aeson (FromJSON (..), withObject, (.:))
+import Data.Function ((&))
+import Data.List.NonEmpty (NonEmpty (..))
+import Data.Text (Text)
+import Prolens (set)
+import Test.Hspec (Spec, describe, it, shouldReturn)
+
+import qualified GitHub as GH
+
+
+querySpecs :: Spec
+querySpecs = describe "Query" $ do
+    it "queries to latest closed issues of 'kowainik/hit-on'" $
+        queryHitonIssues `shouldReturn` Issues
+            [ Issue
+                { issueTitle = "Implement \"git-new\"-like command: `hit hop`"
+                , issueAuthorLogin = "chshersh"
+                }
+            , Issue
+                { issueTitle = "Implement basic CLI interface"
+                , issueAuthorLogin = "vrom911"
+                }
+            ]
+
+queryHitonIssues :: IO Issues
+queryHitonIssues = GH.getGitHubToken "GITHUB_TOKEN" >>= \case
+    Nothing    -> error "No env variable 'GITHUB_TOKEN'"
+    Just token -> GH.queryGitHub token (GH.repositoryToAst issuesQuery)
+
+{-
+query {
+  repository(owner: "kowainik", name: "hit-on") {
+    issues(last: 2, states: [CLOSED], orderBy: {field: CREATED_AT, direction: DESC}) {
+        nodes {
+          title
+          author{
+            login
+          }
+        }
+    }
+  }
+}
+-}
+issuesQuery :: GH.Repository
+issuesQuery = GH.repository
+    ( GH.defRepositoryArgs
+    & set GH.ownerL "kowainik"
+    & set GH.nameL  "hit-on"
+    )
+    $ GH.one
+    $ GH.issues
+        ( GH.defIssuesArgs
+        & set GH.lastL 2
+        & set GH.statesL (GH.one GH.Closed)
+        & set GH.issueOrderL
+            ( Just $ GH.defIssueOrder
+            & set GH.fieldL GH.CreatedAt
+            & set GH.directionL GH.Desc
+            )
+        )
+        ( GH.one
+        $ GH.nodes
+        $ GH.title :| [ GH.author $ GH.one GH.login ]
+        )
```

Comment:
I was assuming that people will export `GitHub` qualified, so decided to give it a go to see how the query looks